sdcm/sct_config.py

```diff
@@ -2970,6 +3048,68 @@ def _verify_rackaware_configuration(self):
+        supported_versions = [
+            v['version'] for v in self.cloud_api_client.get_scylla_versions()['scyllaVersions']
+            if v['newCluster'] == 'ENABLED'
+        ]
+        if (selected_version := self.get('scylla_version')) not in supported_versions:
+            raise ValueError(f"Selected Scylla version '{selected_version}' is not supported by cloud backend.\n"
+                             f"Currently supported versions: {', '.join(supported_versions)}")
+
+        # validate if selected region is supported by the cloud provider
+        provider_id = self.cloud_api_client.cloud_provider_ids[CloudProviderType(cloud_provider.upper())]
+        supported_regions = [
+            r['externalId'] for r in self.cloud_api_client.get_regions(
+                cloud_provider_id=provider_id)['regions']
+        ]
+        region_name = self.get('cloud_region')
+        if region_name not in supported_regions:
+            raise ValueError(f"Selected region '{region_name}' is not supported by cloud provider '{cloud_provider}'.\n"
+                             f"Supported regions for '{cloud_provider}': {', '.join(supported_regions)}")
+
+        # validate if instance types are supported in the selected region
+        region_id = self.cloud_api_client.get_region_id_by_name(
+            cloud_provider_id=provider_id, region_name=region_name)
+        supported_instances = [
+            i['externalId'] for i in self.cloud_api_client.get_instance_types(
+                cloud_provider_id=provider_id, region_id=region_id)['instances']
+        ]
+        if (db_instance_type := self.get('cloud_instance_type_db')) not in supported_instances:
+            raise ValueError(
+                f"Database instance type '{db_instance_type}' is not supported in region '{region_name}' for "
+                f"cloud provider '{cloud_provider}'.\n"
+                f"Supported instance types: {', '.join(supported_instances)}")
+        if (loader_instance_type := self.get('cloud_instance_type_loader')) not in supported_instances:
+            raise ValueError(
+                f"Loader instance type '{loader_instance_type}' is not supported in region '{region_name}' for "
+                f"cloud provider '{cloud_provider}'.\n"
+                f"Supported instance types: {', '.join(supported_instances)}")
+
+        rf = self.get('cloud_replication_factor')
+        n_nodes = self.get('n_db_nodes')
+        if rf is None:
+            self['cloud_replication_factor'] = n_nodes
+        elif rf > n_nodes:
+            raise ValueError(f"cloud_replication_factor ({rf}) cannot be greater than n_db_nodes ({n_nodes})")
+
+        if self.get('n_monitor_nodes') > 0:
+            self.log.warning("Cloud provides built-in monitoring. Setting n_monitor_nodes to 0.")
+            self['n_monitor_nodes'] = 0
+
+        # ensure that client is allowed to access the cloud API
+        client_ip = self.cloud_api_client.client_ip
```

Comment:
this is something we'll need to calculate on run time, i.e. when our loaders would be up.
it's not something we can have upfront, hopefully there API to change it.